pkg/cmd/cli/cmd/newapp.go

```diff
@@ -271,15 +274,52 @@ func setLabels(labels map[string]string, result *newcmd.AppResult) error {
 	return nil
 }
 
+// isInvalidConfigChangeTriggerError returns true if the given error is
+// a validation error that contains 'ConfigChange' in its error message.
+// This error is returned from a server that does not recognize the ConfigChange
+// trigger.
+func isInvalidConfigChangeTriggerError(err error) bool {
+	if !kapierrors.IsInvalid(err) {
+		return false
+	}
+	statusErr, ok := err.(*kapierrors.StatusError)
+	if !ok {
+		return false
+	}
+	return strings.Contains(statusErr.Status().Message, "ConfigChange")
+}
+
+// retryBuildConfig determines if the given error is caused by an invalid ConfigChange trigger
+// error on a BuildConfig. If that is the case, it will remove the trigger from the BuildConfig
+// and return it to be retried.
+func retryBuildConfig(info *resource.Info, err error) runtime.Object {
+	if info.Mapping.Kind == "BuildConfig" && isInvalidConfigChangeTriggerError(err) {
+		bc, ok := info.Object.(*buildapi.BuildConfig)
+		if !ok {
+			return nil
+		}
+		triggers := []buildapi.BuildTriggerPolicy{}
+		for _, t := range bc.Spec.Triggers {
+			if t.Type == buildapi.ConfigChangeBuildTriggerType {
+				continue
+			}
+			triggers = append(triggers, t)
+		}
+		bc.Spec.Triggers = triggers
+		return bc
+	}
+	return nil
+}
+
 func createObjects(f *clientcmd.Factory, out io.Writer, result *newcmd.AppResult) error {
-	// TODO: Validate everything before building
 	mapper, typer := f.Factory.Object()
 	bulk := configcmd.Bulk{
 		Mapper:            mapper,
 		Typer:             typer,
 		RESTClientFactory: f.Factory.RESTClient,
-
-		After: configcmd.NewPrintNameOrErrorAfter(out, os.Stderr),
+		After:             configcmd.NewPrintNameOrErrorAfter(out, os.Stderr),
+		// Retry is used to support previous versions of the API server that will not allow a BuildConfig with a ConfigChange trigger.
```

Comment:
@csrwng this line here also needs to be updated to reflect that it's not particularly about CCT